### PR TITLE
Add functional test for 'markAsFinished'

### PR DESCRIPTION
**Pull request by Mykhailenko**

### CODE REVIEW

File: scheduler/scheduler-server/src/test/java/functionaltests/job/taskkill/TestMarkedAsFinished.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        // wait until normal job is running
+        schedulerHelper.waitForEventJobRunning(normalJobId);
+
+        assertEquals(JobStatus.PAUSED, schedulerHelper.getSchedulerInterface().getJobState(failingJobId).getStatus());
+        assertEquals(JobStatus.RUNNING, schedulerHelper.getSchedulerInterface().getJobState(normalJobId).getStatus());
+
+        // after we mark as finished failed task, failing job should become FINISHED
+        // however, even more important, that other job is continue to run
+        schedulerHelper.getSchedulerInterface().finishInErrorTask(failingJobId.value(), "Error_Task");
+        Thread.sleep(10000);
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/test/java/functionaltests/job/taskkill/TestMarkedAsFinished.java]: MAJOR Remove this use of "Thread.sleep()". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2925)

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: scheduler/scheduler-server/src/test/resources/functionaltests/descriptors/Job_endless.xml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    <description>
+        <![CDATA[ A workflow that executes groovy in JVM. ]]>
+    </description>
+    <taskFlow>
+        <task name="WaitIng_Task" onTaskError="cancelJob">
+            <scriptExecutable>
+                <script>
+                    <code language="groovy">
+                        <![CDATA[
+Thread.sleep(6000000)
```
Comment (fviale) [inline scheduler/scheduler-server/src/test/resources/functionaltests/descriptors/Job_endless.xml]: This sleep is too long. Even if you fix the issue, the task will be running for 6000 seconds !

File: scheduler/scheduler-server/src/test/java/functionaltests/job/taskkill/TestMarkedAsFinished.java [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+        // wait until normal job is running
+        schedulerHelper.waitForEventJobRunning(normalJobId);
+
+        assertEquals(JobStatus.PAUSED, schedulerHelper.getSchedulerInterface().getJobState(failingJobId).getStatus());
+        assertEquals(JobStatus.RUNNING, schedulerHelper.getSchedulerInterface().getJobState(normalJobId).getStatus());
+
+        // after we mark as finished failed task, failing job should become FINISHED
+        // however, even more important, that other job is continue to run
+//        schedulerHelper.getSchedulerInterface().finishInErrorTask(failingJobId.value(), "Error_Task");
+        Thread.sleep(10000);
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/test/java/functionaltests/job/taskkill/TestMarkedAsFinished.java]: MAJOR Remove this use of "Thread.sleep()". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2925)

**Pull request closed without merging**